### PR TITLE
Split `directories` configuration in `twig` and `php` directories

### LINKED ISSUE #13: Improve `directories` in configuration
It's currently not clear that TwigStan scans for both Twig templates and PHP code in `directories`.

Maybe this should be split up:

```neon
parameters:
  twigstan:
    directories:
      twig:
        - templates
      php: 
        - src/Controller
```

### PR DESCRIPTION
**Pull request by raffaelecarelle**
Here is a solution:

As @ruudk said here #13, there are two different types of directories: twig and PHP.

This configuration is injected into the container as a unique array and passed through the finder as before did.

Fixes #13

### CODE REVIEW

File: tests/Application/ContainerFactoryTest.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace TwigStan\Application;
+
+use Nette\DI\Container;
+use PHPUnit\Framework\TestCase;
+
+//TODO Add more test for cover all needed parameters/services
```
Comment (ruudk) [inline tests/Application/ContainerFactoryTest.php:8]:
```suggestion
```

File: tests/Application/ContainerFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        $this->container = $containerFactory->create(sys_get_temp_dir() . '/twigstan');
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        $this->container = null;
+    }
+
```
Comment (ruudk) [inline tests/Application/ContainerFactoryTest.php:27]:
```suggestion
```

File: tests/Application/ContainerFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        self::assertInstanceOf(Container::class, $this->container);
+    }
+
+    public function testDirectoriesHasBeenInjectedCorrectly(): void
+    {
+        $directories = $this->container?->getParameter('twigstan')['directories'];
+
+        self::assertIsArray($directories);
+
+        self::assertCount(4, $directories);
```
Comment (ruudk) [inline tests/Application/ContainerFactoryTest.php:39]: This should test php and twig directories separately.

While at it, also a good moment to test that relative paths become absolute paths.

File: src/Application/ContainerFactory.php [hunk 4]
```diff
@@ -32,9 +32,18 @@ public function create(string $tempDirectory): Container
...
+                $configuration['parameters']['twigstan']['directories']['twig'] ?? [],
+            );
+            $configuration['parameters']['twigstan']['directories']['php'] = array_map(
+                fn(string $directory) => Path::makeAbsolute($directory, Path::getDirectory($this->configurationFile)),
+                $configuration['parameters']['twigstan']['directories']['php'] ?? [],
+            );
+
+            $configuration['parameters']['twigstan']['directories'] = array_merge(
+                $configuration['parameters']['twigstan']['directories']['twig'],
+                $configuration['parameters']['twigstan']['directories']['php'],
```
[the remark this answers is not in the document]
Comment (raffaelecarelle) [reply src/Application/ContainerFactory.php:46]: Ok thats more clear. But what if user pass directory with --paths options? how can know the "scope" (php or twig) of these directories?

[the remark this answers is not in the document]
Comment (ruudk) [reply src/Application/ContainerFactory.php:46]: In those cases, we could search for php+twig files in those directories.

**Comment by raffaelecarelle** [comment]
@ruudk check it out! thank for your review :)

File: src/Finder/FilesFinder.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    /**
+     * @return array<string, SplFileInfo>
+     */
+    public function find(): array
+    {
+        $finder = $this->getFinder();
+
+        $paths = iterator_to_array($finder);
+
+        $paths = array_unique($paths);
```
Comment (ruudk) [inline src/Finder/FilesFinder.php:30]: Is this needed, I don't think so

File: src/Finder/FilesFinder.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+
+    /**
+     * @return array<string, SplFileInfo>
+     */
+    public function find(): array
+    {
+        $finder = $this->getFinder();
+
+        $paths = iterator_to_array($finder);
```
Comment (ruudk) [inline src/Finder/FilesFinder.php:28]:
```suggestion
        return iterator_to_array($finder);
```

File: src/Finder/FilesFinder.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        private array $exclusions,
+    ) {}
+
+
+    /**
+     * @return array<string, SplFileInfo>
+     */
+    public function find(): array
+    {
+        $finder = $this->getFinder();
```
Comment (ruudk) [inline src/Finder/FilesFinder.php:26]: Can you inline this, no need for separate method. Keep it simple

File: src/Application/AnalyzeCommand.php [hunk 8]
```diff
@@ -391,19 +387,31 @@ function ($renderPoints) {
...
 
         $paths = array_unique($paths);
 
+        return $paths;
+    }
+
+    /**
+     * @param list<string> $paths
+     */
+    private function getGenericFinder(array $paths): Finder
```
Comment (ruudk) [inline src/Application/AnalyzeCommand.php:408]: Maybe this can be a separate class as well? Similar to the other finder

**Comment by raffaelecarelle** [comment]
@ruudk done :)

**Review by ruudk — APPROVED** [review]
Thanks @raffaelecarelle 

I made a few changes.

**Pull request merged**